R/symptome_compartment_class.R

```diff
@@ -0,0 +1,226 @@
+#'
+#' Note:
+#' 1) Sum of poulation equals to 1 at any time point.
+#' An error will be reported the sum of initial population is not equal to 1.
+#' 2) At this stage, we only consider disease-caused death cases
+#' calculating mortality.
+#'
+#' model structure:
+#'    ---------------------------(omega)---------------------------------
+#'    |                                                                 |
+#'    |                                                                 |
+#'    --> S --(lam)--> E --(gamma) ------(1-ihr)-------> I ---(nu.i)--> R <--
+#'                                 |                     |                  |
+#'                                 |                 (pdeath.i)             |
+#'                                 |                     |                  |
+#'                                 |            --> CM <--                  |
+#'                                 |            |                           |
+#'                                 |      (pdeath.hsa)                      |
+#'                                 |            |                           |
+#'                                 -- (ihr)---> HA --(nu.hsa)----------------
+
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Class definitions
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+#' @describeIn model_symptome to set up dependent classes for model_symptome
+setClass("ggplot")
+setClass("gg")
+
+#' An S4 class of an basic SEIR model with compartments of different symptomes
+#'
+#' @slot name A string gives the name of the model
+#' @slot initial_population A list of the initial population size of each group
+#' @slot parameters A list of the model parameters
+#' @slot output A dataframe holding the ode simulation output, with columns as population groups and rows as simulation time points
+#' @slot plot_output A ggplot holding the plot of the ode simulation output
+#'
+#' @import deSolve
+#' @import ggplot2
+#' @import reshape2
+#' @import plyr
+#' @import magrittr
+
+setClass(Class = "model_symptome",
+         slots = c(
+           name = "character",
+           initial_population = "list",
+           parameters = "list",
+           output = "list",
+           plot_output = "gg"
+         ),
+         prototype = list(
+           name = NA_character_,
+           initial_population = vector(mode = "list", length = 6) %>%
+             setNames(list("S", "E", "I", "HA", "R", "CM")),
+           parameters = vector(mode = "list", length = 6) %>%
+             setNames(list("lam", "gamma", "omega", "ihr", "nu", "pdeath")),
+           output = data.frame(),
+           plot_output = ggplot()
+         ))
+              
+#' @describeIn model_symptome set initial population sizes (in fraction) and parameters
+#'
+#' @param object An object of class model_symptome
+#' @param S Numeric, initial population of the susceptible
```

Comment:
maybe "initial fraction of the population that is susceptible" or something along those lines is a bit more consistent with a normalized population. If not, I think "initial population of susceptible" might be slightly better English.